azurerm/internal/services/resource/data_source_resource_groups_test.go

```diff
@@ -0,0 +1,72 @@
+package resource_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance/check"
+)
+
+type ResourceGroupsDataSource struct{}
+
+func TestAccDataSourceResourceGroups_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_resource_groups", "test")
+
+	data.DataSourceTest(t, []resource.TestStep{
+		{
+			Config: ResourceGroupsDataSource{}.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("resource_groups.0.id").Exists(),
+				check.That(data.ResourceName).Key("resource_groups.0.name").Exists(),
+				check.That(data.ResourceName).Key("resource_groups.0.type").Exists(),
+				check.That(data.ResourceName).Key("resource_groups.0.location").Exists(),
+				check.That(data.ResourceName).Key("resource_groups.0.subscription_id").Exists(),
+				check.That(data.ResourceName).Key("resource_groups.0.tenant_id").Exists(),
+			),
+		},
+	})
+}
+
+func TestAccDataSourceResourceGroups_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_resource_groups", "test")
+
+	data.DataSourceTest(t, []resource.TestStep{
+		{
+			Config: ResourceGroupsDataSource{}.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("resource_groups.0.id").Exists(),
+				check.That(data.ResourceName).Key("resource_groups.0.name").Exists(),
+				check.That(data.ResourceName).Key("resource_groups.0.type").Exists(),
+				check.That(data.ResourceName).Key("resource_groups.0.location").Exists(),
+				check.That(data.ResourceName).Key("resource_groups.0.subscription_id").Exists(),
+				check.That(data.ResourceName).Key("resource_groups.0.tenant_id").Exists(),
+			),
+		},
+	})
+}
+
+func (d ResourceGroupsDataSource) basic(data acceptance.TestData) string {
+	return `
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_resource_groups" "test" {}
+`
+}
+
+func (d ResourceGroupsDataSource) complete(data acceptance.TestData) string {
+	return `
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_client_config" "current" {
+}
+
+data "azurerm_resource_groups" "test" {
+  filter_by_subscription_id = [data.azurerm_client_config.current.subscription_id]
+}
+`
+}
```

Comment:
Would it make sense to extend the test to create multiple resource groups and check for at least correct amount returned in both basic and complete?

Reply:
Hi @favorretti Yes it would, but I am not sure how to do that correctly: I need to make sure that one comes after the other, i.e. that first the resource groups are deployed and then the data source is used...

Reply:
Hi @favoretti tried to address your review. Can you please check again? :)